libflux/flux-core/src/parser/mod.rs

```diff
@@ -363,18 +377,28 @@ impl<'input> Parser<'input> {
         let start_pos = ast::Position::from(&self.peek().start_pos);
         let mut end = ast::Position::invalid();
 
-        // Parse attributes at the beginning of the file.
-        let attributes = self.parse_attribute_list();
+        // Parse inner attributes at the beginning of the file and hand them off to the first
+        // clause, declaration or statement that exists
+        let inner_attributes = self.parse_attribute_inner_list();
```

Comment:
There is no such thing as "inner" or "outer" attributes now, right? I wonder if it will be clearer to omit `inner_` here?

Reply:
I left the `inner` vocabulary because its still accurate in that these attributes can only apply to an inner context. Unlike before this change the attributes could apply to either an inner or outer context. Even if we never add an explicit outer syntax it makes it more specific that its about the inner context only.